File: 백준/Bronze/4153. 직각삼각형/직각삼각형.cc
#include <iostream>
#include <algorithm>
#include <queue>
#include <tuple>

using namespace std;


int main() {

    vector<int> arr(3);
    while(true){
        for(int i = 0; i < 3; i++) cin >> arr[i];
        if(arr[0] == 0 && arr[1] == 0 && arr[2] == 0) break;
        sort(arr.begin(),arr.end());
        if(arr[0]*arr[0]+arr[1]*arr[1] == arr[2]*arr[2]) cout << "right" << '\n';
        else cout << "wrong" << '\n';
    }


    return 0;
}

File: 백준/Bronze/4153. 직각삼각형/직각삼각형.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        Scanner in = new Scanner(System.in);
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st;

        while (true) {
            ArrayList<Integer> arr = new ArrayList<>();
            st = new StringTokenizer(br.readLine());
            while (st.hasMoreTokens()) {
                arr.add(Integer.parseInt(st.nextToken()));
            }
            if (arr.get(0).equals(arr.get(1)) &&
                    arr.get(1).equals(arr.get(2)) && arr.get(0) == 0) break;
            Collections.sort(arr);
            if (Math.pow(arr.get(0), 2) + Math.pow(arr.get(1), 2) == Math.pow(arr.get(2), 2)){
                System.out.println("right");
            }
            else System.out.println("wrong");
        }
    }
}